### PR TITLE
Add support for the include directive in XCompose files

### PR DESCRIPTION
**Pull request by lf-**
We also fix an issue with excess space at the start of lines stopping
comments being recognized.

### CODE REVIEW

**Review by fujiwarat — CHANGES_REQUESTED** [review]

**Comment by fujiwarat** [comment]
@lf- Do you have a chance to update the patch?

**Comment by lf-** [comment]
@fujiwarat Yes, I plan to work on the patch this week during my reading break. Sorry I took a while to get to it, school has been very difficult.

File: src/ibuscomposetable.c [hunk 1]
```diff
@@ -243,10 +244,58 @@ parse_compose_sequence (IBusComposeData *compose_data,
...
+    const gchar *i = include_path;
+    gchar *o = out;
+
+    while (*i && o < out_lastchar) {
+        // expand sequence
+        if (*i == '%') {
+            switch (*(i+1)) {
+            // $HOME
+            case 'H':
+                o = out + g_strlcat(out, getenv("HOME"), PATHLEN_MAX);
```
[the remark this answers is not in the document]
Comment (lf-) [reply src/ibuscomposetable.c:260]: I think it is correct as it originally was: [`g_strlcat`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strlcat) returns the length of the concatenated string or if it ran out of space, the max length - 1. Thus we want to add the value returned to the pointer to the start of the string so `o` becomes a pointer to the character we want to write next.

**Comment by lf-** [comment]
OK I believe I've addressed all the feedback. I've force pushed the changed version.

**Comment by fujiwarat** [comment]
Thank you for the patch.
I added another patch not to include the same compose file and the system EN compose file.

**Review by fujiwarat — COMMENTED** [review]
Done

**Pull request closed without merging**